optuna/importance/_base.py

```diff
@@ -132,3 +137,51 @@ def _check_evaluate_args(completed_trials: List[FrozenTrial], params: Optional[L
                     "Study must contain completed trials with all specified parameters. "
                     "Specified parameters: {}.".format(params)
                 )
+
+
+def _split_distributions(
+    distributions: Dict[str, BaseDistribution]
+) -> Tuple[Dict[str, BaseDistribution], Dict[str, BaseDistribution]]:
+    non_single_distributions = {
+        name: dist for name, dist in distributions.items() if not dist.single()
+    }
+    single_distributions = {name: dist for name, dist in distributions.items() if dist.single()}
+    return (non_single_distributions, single_distributions)
+
+
+def _get_filtered_trials(
+    study: Study, params: Collection[str], target: Callable[[FrozenTrial], float]
+) -> List[FrozenTrial]:
+    trials = study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))
+    return [
+        trial
+        for trial in trials
+        if set(params) <= set(trial.params) and numpy.isfinite(float(target(trial)))
+    ]
+
+
+def _feature_importances_to_param_importances(
+    feature_importances: numpy.ndarray, trans: _SearchSpaceTransform
+) -> numpy.ndarray:
+    param_importances = numpy.zeros(trans.num_params)
+    numpy.add.at(param_importances, trans.encoded_column_to_column, feature_importances)
+    return param_importances
```

Comment:
I think this logic is not so common and a bit domain specific as it depends feature_importances representation of scikit-learn. We cannot make this logic DRY by moving this logic from `optuna/importance/_mean_decrease_impurity.py`.

How about moving this logic to `optuna/importance/_mean_decrease_impurity.py`?

Reply:
I think this logic is as common as `_SearchSpaceTransform`, since what this function does is basically un-transform the importance values computed on the per-feature basis.

Reply:
> I think this logic is as common as _SearchSpaceTransform

At least, this logic looks not duplicated now. This change would rather increase the lines of code than before. I think we should wait to try refactoring until we can gain the benefit of DRY. It helps us to avoid premature abstractions. What do you think?